### ISSUE TITLE
No UiMapID for (nil) :405Nataka Longhorn [C]

### ISSUE THREAD

**Issue by Djfe** [AeroScripts/QuestieDev#1984]
## Bug description

A bunch of add-ons threw lua errors during my raiding tonight (I think in MC)
I'm not sure if this bug was caused by questie itself, but I'll leave the trace here:
```
1x Questie\Modules\Map\QuestieMap.lua:426: No UiMapID for (nil) :405Nataka Longhorn
[C]: in function `error'
Questie\Modules\Map\QuestieMap.lua:426: in function `DrawWorldIcon'
Questie\Modules\Quest\QuestieQuest.lua:1376: in function `DrawAvailableQuest'
Questie\Modules\Quest\QuestieQuest.lua:1400: in function `DrawAllAvailableQuests'
...rfaceQuestie\Modules\QuestieEventHandler.lua:303: in function <...rfaceQuestie\Modules\QuestieEventHandler.lua:297>
[C]: ?
...las\Libs\CallbackHandler-1.0\CallbackHandler-1.0-7.lua:29: in function <...las\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:25>
...las\Libs\CallbackHandler-1.0\CallbackHandler-1.0-7.lua:64: in function `Fire'
...\common\Wildpants\libs\AceEvent-3.0\AceEvent-3.0-4.lua:120: in function <...\common\Wildpants\libs\AceEvent-3.0\AceEvent-3.0.lua:119>
```

## Questie version

5.10.3

**Comment by Djfe** [comment]
Another bug, same raid a bit later (maybe also in the same minute?)
```
1x ...ceQuestie\Modules\Libs\QuestieSerializer.lua:124: table index is nil
...ceQuestie\Modules\Libs\QuestieSerializer.lua:124: in function <...ceQuestie\Modules\Libs\QuestieSerializer.lua:113>
(tail call): ?
(tail call): ?
...ceQuestie\Modules\Libs\QuestieSerializer.lua:132: in function <...ceQuestie\Modules\Libs\QuestieSerializer.lua:129>
(tail call): ?
(tail call): ?
...ceQuestie\Modules\Libs\QuestieSerializer.lua:120: in function <...ceQuestie\Modules\Libs\QuestieSerializer.lua:113>
(tail call): ?
(tail call): ?
...ceQuestie\Modules\Libs\QuestieSerializer.lua:120: in function <...ceQuestie\Modules\Libs\QuestieSerializer.lua:113>
(tail call): ?
(tail call): ?
...ceQuestie\Modules\Libs\QuestieSerializer.lua:120: in function <...ceQuestie\Modules\Libs\QuestieSerializer.lua:113>
(tail call): ?
(tail call): ?
...ceQuestie\Modules\Libs\QuestieSerializer.lua:120: in function <...ceQuestie\Modules\Libs\QuestieSerializer.lua:113>
...ceQuestie\Modules\Libs\QuestieSerializer.lua:327: in function `Deserialize'
...faceQuestie\Modules\Network\QuestieComms.lua:396: in function <...faceQuestie\Modules\Network\QuestieComms.lua:393>
[C]: ?
...las\Libs\CallbackHandler-1.0\CallbackHandler-1.0-7.lua:29: in function <...las\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:25>
...las\Libs\CallbackHandler-1.0\CallbackHandler-1.0-7.lua:64: in function `Fire'
...faceBugSack\Libs\AceComm-3.0\AceComm-3.0-12.lua:218: in function `OnReceiveMultipartLast'
...faceBugSack\Libs\AceComm-3.0\AceComm-3.0-12.lua:256: in function <...faceBugSack\Libs\AceComm-3.0\AceComm-3.0.lua:246>
```

**Comment by BreakBB** [comment]
The first one sounds like something went wrong when initialized the database and zones. Because zoneID 405 = Desolace which is in our database.

The second error happened because Blizzard add a limit to the addon communication API without saying anything about it. We're working on a fix for it.

**Comment by BreakBB** [comment]
The refactoring of 93a51ca65d417b24f7b90d38a0a8a3381e21b826 should help to prevent such zoneID errors and will work more reliable. The changes will be included in the upcoming release of v6.0.0
